java-manta-client/src/test/java/com/joyent/manta/client/multipart/EncryptedMultipartManagerTest.java

```diff
@@ -0,0 +1,212 @@
+
+        MantaHttpHeaders actualHeaders = MantaObjectMapper.INSTANCE
+                .readValue(actualUpload.getHeaders(), MantaHttpHeaders.class);
+
+        Assert.assertEquals(actualHeaders, headers,
+                "Headers were stored correctly");
+
+        {
+            Cipher cipher = cipherDetails.getCipher();
+            byte[] iv = upload.getEncryptionState().getEncryptionContext().getCipher().getIV();
+            cipher.init(Cipher.ENCRYPT_MODE, secretKey, cipherDetails.getEncryptionParameterSpec(iv));
+
+            byte[] assumedCiphertext = cipher.doFinal(expected.getBytes("US-ASCII"));
+            byte[] contents = FileUtils.readFileToByteArray(upload.getWrapped().getContents());
+            byte[] actualCiphertext = Arrays.copyOf(contents,
+                    contents.length - cipherDetails.getAuthenticationTagOrHmacLengthInBytes());
+
+            try {
+                AssertJUnit.assertEquals(assumedCiphertext, actualCiphertext);
+            } catch (AssertionError e) {
+                System.err.println("expected: " + Hex.encodeHexString(assumedCiphertext));
+                System.err.println("actual  : " + Hex.encodeHexString(actualCiphertext));
+                throw e;
+            }
+        }
+
+        EncryptionContext encryptionContext = upload.getEncryptionState().getEncryptionContext();
+
+        MantaHttpHeaders responseHttpHeaders = new MantaHttpHeaders();
+        responseHttpHeaders.setContentLength(actualUpload.getContents().length());
+        responseHttpHeaders.put(MantaHttpHeaders.ENCRYPTION_HMAC_TYPE,
+                encryptionContext.getCipherDetails().getAuthenticationHmac().getAlgorithm());
+        responseHttpHeaders.put(MantaHttpHeaders.ENCRYPTION_IV,
+                Base64.getEncoder().encodeToString(encryptionContext.getCipher().getIV()));
+        responseHttpHeaders.put(MantaHttpHeaders.ENCRYPTION_KEY_ID,
+                cipherDetails.getCipherId());
+
+        MantaObjectResponse response = new MantaObjectResponse(path,
+                responseHttpHeaders, metadata);
+
+
+        CloseableHttpResponse httpResponse = mock(CloseableHttpResponse.class);
+
+        try (InputStream fin = new FileInputStream(actualUpload.getContents());
+             EofSensorInputStream eofIn = new EofSensorInputStream(fin, null);
+             MantaObjectInputStream objIn = new MantaObjectInputStream(response, httpResponse, eofIn);
+             InputStream in = new MantaEncryptedObjectInputStream(objIn, cipherDetails, secretKey, true)) {
+            String actual = IOUtils.toString(in, "UTF-8");
+
+            Assert.assertEquals(actual, expected);
+        }
+    }
+
+    // TEST UTILITY METHODS
+
+    private SettableConfigContext<BaseChainedConfigContext> testConfigContext(SecretKey key) {
+        StandardConfigContext settable = new StandardConfigContext();
+        settable.setMantaUser("test");
+
+        final String privateKey = "-----BEGIN RSA PRIVATE KEY-----\n" +
```

Comment:
Add a comment "key generated just for testing", so that some automatic audit tool doesn't freak out later.